sycl/include/CL/sycl/buffer.hpp

```diff
@@ -134,13 +134,10 @@ class buffer {
     if (b.is_sub_buffer())
       throw cl::sycl::invalid_object_error(
           "Cannot create sub buffer from sub buffer.");
-    // TODO: SYCL CTS are failed because not appropriate offset and range
-    // are passed to this constructor which causes a throw of exception.
-    // Some changes need to be done there.
-    // if (isOutOfBounds(baseIndex, subRange, b.Range))
-    //   throw cl::sycl::invalid_object_error("Out-of-bounds size");
-    // if (!isContiguousRegion(baseIndex, subRange, b.Range))
-    //   throw cl::sycl::invalid_object_error("Non-contiguous region");
+    if (isOutOfBounds(baseIndex, subRange, b.Range))
+      throw cl::sycl::invalid_object_error("Out-of-bounds size");
```

Comment:
I'm not an expert, but "Out-of-bounds size" sounds weird.
What about "Requested sub-buffer size exceeds the size of the parent buffer"

Reply:
Thanks for noticing. Fixed.